src/shogun/distributions/classical/GaussianDistributionNew.cpp

```
@@ -0,0 +1,297 @@
+	m_cov(0,0)=cov;
+
+	m_cov_type="Float";
+}
+
+CGaussianDistributionNew::CGaussianDistributionNew(SGVector<float64_t> mean,
+		SGVector<float64_t> cov, bool cov_is_cholesky) :
+				CProbabilityDistribution(mean.vlen)
+{
+	REQUIRE(mean.vlen==cov.vlen,
+			"Mean dimension (%d) and covariance dimension (%d) do not match\n",
+			mean.vlen, cov.vlen);
+
+	init();
+
+	m_mean=mean;
+
+	m_cov=cov.convert_to_matrix(cov,1,cov.vlen,false);
+
+	m_cov_type="Vector";
+}
+
+CGaussianDistributionNew::CGaussianDistributionNew(SGVector<float64_t> mean,
+		SGMatrix<float64_t> cov, bool cov_is_cholesky) :
+				CProbabilityDistribution(mean.vlen)
+{
+	REQUIRE(cov.num_rows==cov.num_cols, "Covariance must be square but is "
+			"%dx%d\n", cov.num_rows, cov.num_cols);
+	REQUIRE(mean.vlen==cov.num_cols,
+			"Mean dimension (%d) and covariance dimension (%d) do not match\n",
+			mean.vlen, cov.num_cols);
+
+	init();
+
+	m_mean=mean;
+
+	if (!cov_is_cholesky)
+	{
+		compute_cholesky(cov);
+	}
+	else
+		m_cov=cov;
+
+	m_cov_type="Matrix";
+}
+
+CGaussianDistributionNew::~CGaussianDistributionNew()
+{
+
+}
+
+void CGaussianDistributionNew::set_mean(SGVector<float64_t> mean)
+{
+	m_mean=mean;
+}
+
+void CGaussianDistributionNew::set_cov(float64_t cov)
+{
+	m_cov=SGMatrix<float64_t>(1,1);
+	m_cov(0,0)=cov;
+
+	m_cov_type="Float";
+}
+
+void CGaussianDistributionNew::set_cov(SGVector<float64_t> cov)
+{
+	m_cov=cov.convert_to_matrix(cov,1,cov.vlen,false);
+
+	m_cov_type="Vector";
+}
+
+void CGaussianDistributionNew::set_cov(SGMatrix<float64_t> cov)
+{
+	m_cov=cov;
+
+	m_cov_type="Matrix";
+}
+
+SGVector<float64_t> CGaussianDistributionNew::get_mean() const
+{
+	return m_mean;
+}
+
+float64_t CGaussianDistributionNew::get_cov_spherical() const
+{
+	return *m_cov.data();
+}
+
+SGVector<float64_t> CGaussianDistributionNew::get_cov_diag() const
+{
+	return m_cov.get_row_vector(0);
+}
+
+SGMatrix<float64_t> CGaussianDistributionNew::get_cov_full() const
+{
+	return m_cov;
```

Should throw an error if m_cov is not COV_FULL

This is wrong as the m_cov stores a cholesky factor. So you need to allocate a new matrix here, compute LL^T, and return that.  Also the documentation should state that this gives a copy.

I also suggest to add another methods get_cov_full_cholesky() which returns m_cov if m_cov_type == COV_FULL